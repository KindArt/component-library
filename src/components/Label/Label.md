Standard:

```jsx
<Label content="This is the content" />
```
