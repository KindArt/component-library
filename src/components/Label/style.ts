import styled, { css } from 'styled-components';
import { Colours } from '../../constants/styles';

export const LabelWrapper = styled.span`
  font-weight: 500;
  font-size: 0.75rem;
  color: ${({ theme }) => theme.label.colour};
  line-height: 1.125rem;
  text-transform: uppercase;
  & + * {
    margin-top: 0.5rem;
  }

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

LabelWrapper.defaultProps = {
  theme: {
    label: {
      colour: `${Colours.TEXT_SECONDARY}`,
    },
  },
};
