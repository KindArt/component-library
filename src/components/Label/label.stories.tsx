import React from 'react';
import Label from '.';

export default {
  component: Label,
  title: 'Label',
};

export const normal = () => <Label content="This is a label" />;
