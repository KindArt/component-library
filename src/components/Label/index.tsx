import React, { FC, HTMLProps } from 'react';
import { LabelWrapper } from './style';

interface ILabel extends HTMLProps<HTMLLabelElement> {
  id?: string;
  content: string;
  testIds?: {
    wrapper?: string;
  };
  classOverrides?: {
    wrapper?: string;
  };
}

const componentName = 'Label';

const Label: FC<ILabel> = ({ id, content, testIds, classOverrides, ...props }) => {
  const tidPrefix = id || componentName;
  const tid = {
    wrapper: (testIds && testIds.wrapper) || 'wrapper',
  };

  return (
    <LabelWrapper
      className={classOverrides && classOverrides.wrapper}
      data-testid={`${tidPrefix}-${tid.wrapper}`}
      {...(props as any)}
    >
      {content}
    </LabelWrapper>
  );
};

export default Label;
