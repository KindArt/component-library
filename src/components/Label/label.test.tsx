import React from 'react';
import { shallow } from 'enzyme';
import Label from '.';

describe('<Label />', () => {
  let wrapper: any;
  beforeEach(() => {
    wrapper = shallow(<Label content="This is a label" />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Test Ids', () => {
    it('should set a custom test id for wrapper', () => {
      wrapper.setProps({ testIds: { wrapper: 'customWrapper' } });
      const actual = wrapper.find('[data-testId="Label-customWrapper"]').length;
      const expected = 1;
      expect(actual).toEqual(expected);
    });
  });

  describe('Class Overrides', () => {
    it('should override the wrapper class', () => {
      wrapper.setProps({ classOverrides: { wrapper: 'override' } });
      const actual = wrapper.find('[data-testId="Label-wrapper"]').hasClass('override');
      expect(actual).toBeTruthy();
    });
  });
});
