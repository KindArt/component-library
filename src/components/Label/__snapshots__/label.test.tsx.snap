// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render 1`] = `
<styled.label
  data-testid="Label-wrapper"
  theme={
    Object {
      "label": Object {
        "colour": "#9EA0A5",
      },
    }
  }
>
  This is a label
</styled.label>
`;
